elastic/kibana
# [Event Log] Fixing call to update index alias to hidden

Resolves https://github.com/elastic/kibana/issues/121298

## Summary

When upgrading from <= 7.15 versions of Kibana, the event log will try to update existing event log indices to `hidden`. Part of this process involves setting existing index aliases to `is_hidden`. It turns out if you have one alias pointing to multiple indices, we need to update the alias settings in a single command instead of consecutive commands as we were doing previously.

For example, if our 7.15 event log aliases look like this:

```
{
	".kibana-event-log-7.15.2-000003": {
		"aliases": {
			".kibana-event-log-7.15.2": {
				"is_write_index": true
			}
		}
	},
	".kibana-event-log-7.15.2-000002": {
		"aliases": {
			".kibana-event-log-7.15.2": {
				"is_write_index": false
			}
		}
	},
	".kibana-event-log-7.15.2-000001": {
		"aliases": {
			".kibana-event-log-7.15.2": {
				"is_write_index": false
			}
		}
	}
}
```

Previously, we were issuing 3 calls to the update alias API, each one looking like this: 

```
{
	actions: [{
		add: {
			index: <indexName>,
			alias: '.kibana-event-log-7.15.2',
			is_hidden: true,
		},
	}, ],
}
```

We actually need to be issuing 1 call to the update alias API that looks like this:

```
{
	"actions": [{
			"add": {
				"is_write_index": false,
				"index": ".kibana-event-log-7.15.2-000002",
				"alias": ".kibana-event-log-7.15.2",
				"is_hidden": true
			}
		},
		{
			"add": {
				"is_write_index": true,
				"index": ".kibana-event-log-7.15.2-000003",
				"alias": ".kibana-event-log-7.15.2",
				"is_hidden": true
			}
		},
		{
			"add": {
				"is_write_index": false,
				"index": ".kibana-event-log-7.15.2-000001",
				"alias": ".kibana-event-log-7.15.2",
				"is_hidden": true
			}
		}
	]
}
```

This PR updates to do this correctly.

## To Verify

1. Change the Event log ILM policy to max_docs: 1 (to get ILM to roll over faster). Run 7.15 (or previous version) and let some rules/actions run to generate some event log docs and get multiple event log indices.
2. Run `main` using the previous data and notice that there is an error in the logs when updating the index aliases that cause the aliases not to get updated to `hidden`.
3. Run this branch using the previous data and notice no errors in the logs that verify that the index aliases are all updated to hidden.

### Checklist
